src/cbmc/bmc.cpp

```diff
@@ -296,9 +297,8 @@ void bmct::get_memory_model()
     memory_model=util_make_unique<memory_model_psot>(ns);
   else
   {
-    error() << "Invalid memory model " << mm
-            << " -- use one of sc, tso, pso" << eom;
-    throw "invalid memory model";
+    throw invalid_user_input_exceptiont(
+      "Invalid parameter " + mm, "--mm", "values of sc, tso, pso");
```

Comment:
I'm struggling to piece this together: what sort of sentence is going to be formed using "values of ..."?

Reply:
Yes, but in order to figure out what to put in those parameters I will almost always have to look at the code generating this string. I don't think it's particularly obvious what (not) to put in the arguments. It would already be an improvement if the "Try " were not automatically included so that the argument is a full sentence?